sklearn/utils/fixes.py

```diff
@@ -39,6 +40,59 @@
 except ImportError:  # SciPy < 1.8
     from scipy.optimize.linesearch import line_search_wolfe2, line_search_wolfe1  # type: ignore  # noqa
 
+if sp_version >= parse_version("1.9"):
+    from scipy.stats import mode
+else:
+
+    def mode(a, axis=0, nan_policy="propagate"):
+        """Return an array of the modal (most common) value in the passed array.
+
+        If there is more than one such value, only the smallest is returned.
+        The bin-count for the modal bins is also returned.
+
+        Parameters
+        ----------
+        a : array_like
+            n-dimensional array of which to find mode(s).
+        axis : int or None, optional
+            Axis along which to operate. Default is 0. If None, compute over
+            the whole array `a`.
+        nan_policy : {'propagate', 'raise', 'omit'}, optional
+            Defines how to handle when input contains nan.
+            The following options are available (default is 'propagate'):
+            * 'propagate': returns nan
+            * 'raise': throws an error
+            * 'omit': performs the calculations ignoring nan values
+
+        Returns
+        -------
+        mode : ndarray
+            Array of modal values.
+        count : ndarray
+            Array of counts for each mode.
+
+        Examples
+        --------
+        >>> import numpy as np
+        >>> a = np.array([[6, 8, 3, 0],
+        ...               [3, 2, 1, 7],
+        ...               [8, 1, 8, 4],
+        ...               [5, 3, 0, 5],
+        ...               [4, 7, 5, 9]])
+        >>> from sklearn.utils.fixes import mode
+        >>> mode(a)
+        ModeResult(mode=array([3, 1, 0, 0]), count=array([1, 1, 1, 1]))
+        To get mode of whole array, specify ``axis=None``:
+        >>> mode(a, axis=None)
+        ModeResult(mode=3, count=3)
+        """
```

Comment:
This example will need to behave differently based on what version of scipy people run, correct? So either way it'll fail CI docstring test right?

Reply:
Uhm no. The idea is that `sklearn.utils.fixes.mode` will always behave like `scipy>=1.9`. So we either import it or redefine it and reduce the dimension.

Reply:
Yup sorry I hat a short-circuit there for some reason :P